.github/workflows/codeql.yml

```diff
@@ -16,32 +16,45 @@
 name: CodeQL
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [ main ]
+    branches: [main]
 
 permissions:
   contents: read
   security-events: write
 
 jobs:
+  go_version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
+      - name: Get Go version directive from go.mod as major.minor
+        id: list
+        run: |
+          go list -m -f 'version={{.GoVersion}}' | cut -d. -f-2 >>"${GITHUB_OUTPUT}"
+    outputs:
+      version: ${{ steps.list.outputs.version }}
   analyze:
     name: Analyze
+    needs:
+      - go_version
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
-        language: [ 'go' ]
+        language: ["go"]
     steps:
       - name: Checkout repository
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
       - name: Set up Go
         uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
         with:
-          go-version-file: go.mod
+          go-version: "${{ needs.go_version.outputs.version }}.x"
```

Comment:
in the action, we don't need to set the .x, only major.minor is fine

Reply:
My understanding from setup-go is that `1.22` is analogous to `1.22.0` whereas `1.22.x` is analogous to "use latest patch version" and so (as of today) would get `1.22.9` which is what we desire for testing?

Reply:
with the check-latest it will pick up the latest one, but i am okay to keep the .x not a big thing

Reply:
ah yes you're right, gave it a quick test:

```
Run actions/setup-go@v5
  with:
    go-version: 1.22
    check-latest: true
    token: ***
    cache: true
Setup go version spec 1.22
Attempting to resolve the latest version from the manifest...
matching 1.22...
Resolved as '1.22.9'
Attempting to download 1.22.9...
matching 1.22.9...
Acquiring 1.22.9 from https://github.com/actions/go-versions/releases/download/1.22.9-11715914133/go-1.22.9-linux-x64.tar.gz
```